libtransmission/rpc-server.cc

```diff
@@ -51,12 +58,49 @@ using namespace std::literals;
 
 static char constexpr MyName[] = "RPC Server";
 
+static char constexpr TrUnixSocketPrefix[] = "unix:";
+
+/* The maximum size of a unix socket path is defined per-platform based on sockaddr_un.sun_path.
+ * On Windows the fallback is the length of the unix prefix. Subtracting one at the end is for
+ * double counting null terminators from sun_path and TrUnixSocketPrefix. */
+#ifdef _WIN32
+auto inline constexpr TrUnixAddrStrLen = size_t{ sizeof(TrUnixSocketPrefix) };
+#else
+auto inline constexpr TrUnixAddrStrLen = size_t{ sizeof(((struct sockaddr_un*)nullptr)->sun_path) + sizeof(TrUnixSocketPrefix) -
+                                                 1 };
+#endif
+
+enum tr_rpc_address_type
+{
+    TR_RPC_AF_INET,
+    TR_RPC_AF_INET6,
+    TR_RPC_AF_UNIX
+};
+
+struct tr_rpc_address
+{
+    tr_rpc_address_type type;
+    union
+    {
+        struct in6_addr;
+        struct in_addr;
```

Comment:
These two lines forward-declare two types, they don't actually influence the size of the union.

Reply:
I'm not quite sure what you mean by this.

Reply:
They need a variable name, as well as the type, to be fields in the union.

```c++
          struct in6_addr addr6;
          struct in_addr addr4;
```

Reply:
Wow, I completely missed this